From d17772fa2af152028daec45f856445edca6a5212 Mon Sep 17 00:00:00 2001
From: Ricardo Salveti <ricardo@opensourcefoundries.com>
Date: Wed, 9 May 2018 16:42:01 -0300
Subject: [PATCH] sota_tools: always set cacerts if specified by the user

ca_certs wasn't set for every authentication method even when specified
by the user via command line (garage-push --cacert).

Signed-off-by: Ricardo Salveti <ricardo@opensourcefoundries.com>
---
 src/sota_tools/authenticate.cc | 10 ++++++----
 1 file changed, 6 insertions(+), 4 deletions(-)

diff --git a/src/sota_tools/authenticate.cc b/src/sota_tools/authenticate.cc
index 5b58d6c7..f2856f56 100644
--- a/src/sota_tools/authenticate.cc
+++ b/src/sota_tools/authenticate.cc
@@ -27,9 +27,6 @@ int authenticate(const string &cacerts, const ServerCredentials &creds, TreehubS
       } else {
         LOG_INFO << "Skipping Authentication";
       }
-      // Set ca certificate path, because curl embeds the path to ca-certs that it was built with
-      // and this breaks under bitbake when sharing sstate cache between machines
-      treehub.ca_certs(cacerts);
       break;
     }
     case CERT: {
@@ -37,7 +34,6 @@ int authenticate(const string &cacerts, const ServerCredentials &creds, TreehubS
       break;
     }
     case AUTH_NONE:
-      treehub.ca_certs(cacerts);  // Setup ca certificate because curl by default check ca certs
       break;
 
     default: {
@@ -45,6 +41,12 @@ int authenticate(const string &cacerts, const ServerCredentials &creds, TreehubS
       return EXIT_FAILURE;
     }
   }
+
+  if (cacerts != "") {
+    LOG_INFO << "Setting cacerts to " << cacerts;
+    treehub.ca_certs(cacerts);
+  }
+
   treehub.root_url(creds.GetOSTreeServer());
   treehub.repo_url(creds.GetRepoUrl());
 
-- 
2.17.0

